pyccl/core.py

```diff
@@ -1148,3 +933,278 @@ def __init__(self, **kwargs):
+                                                        status)
+            check(status, self)
+
+        # Growth
+        if has_dz:
+            if not isinstance(growth, dict):
+                raise TypeError("`growth` must be a dictionary.")
+            if (('a' not in growth) or ('growth_factor' not in growth) or
+                    ('growth_rate' not in growth)):
+                raise ValueError("`growth` must contain keys "
+                                 "'a', 'growth_factor' and 'growth_rate'")
+            a = growth['a']
+            dz = growth['growth_factor']
+            fz = growth['growth_rate']
+            # Check that input arrays have the same size.
+            if not (a.shape == dz.shape
+                    == fz.shape):
+                raise ValueError("Input arrays must have the same size.")
+            # Check that a_array_grth is a monotonically increasing array.
+            if not np.array_equal(a,
+                                  np.sort(a)):
+                raise ValueError("Input scale factor array is not "
+                                 "monotonically increasing.")
+            # Check that the last element of a_array_grth is 1:
+            if np.abs(a[-1]-1.0) > 1e-5:
+                raise ValueError("The last element of the input scale factor"
+                                 "array must be 1.0.")
+            status = 0
+            status = lib.cosmology_growth_from_input(self.cosmo,
+                                                     a, dz, fz,
+                                                     status)
+            check(status, self)
+
+        # Linear power spectrum
+        if has_pklin:
+            if not isinstance(pk_linear, dict):
+                raise TypeError("`pk_linear` must be a dictionary")
+            if (('delta_matter_x_delta_matter' not in pk_linear) or
+                    ('a' not in pk_linear) or ('k' not in pk_linear)):
+                raise ValueError("`pk_linear` must contain keys 'a', 'k' "
+                                 "and 'delta_matter_x_delta_matter' "
+                                 "(at least)")
+            na = len(pk_linear['a'])
+            nk = len(pk_linear['k'])
+            lk = np.log(pk_linear['k'])
+            pk_names = [key for key in pk_linear if key not in ('a', 'k')]
+            for n in pk_names:
+                qs = n.split('_x_')
+                if len(qs) != 2:
+                    raise ValueError("Power spectrum label %s could " % n +
+                                     "not be parsed. Label must be of the " +
+                                     "form 'q1_x_q2'")
+                pk = pk_linear[n]
+                if pk.shape != (na, nk):
+                    raise ValueError("Power spectrum %s has shape " % n +
+                                     str(pk.shape) + " but shape " +
+                                     "(%d, %d) was expected." % (na, nk))
```

Comment:
Did we want to check that a and k are properly ordered here?

Reply:
do you mean check that the arrays are sorted? Or check if `pk.shape == (nk, na)` and, if so, flip it?

Reply:
Both maybe?

Reply:
nvm per discussion above.